packages/lesswrong/server/dialogues/cron.ts

```diff
@@ -21,3 +26,47 @@ addCronJob({
     })
   }
 });
+
+async function checkActiveUserSession(userId:string, documentId:string) {
+  const ckEditorDocumentId = getCKEditorDocumentId(documentId, userId, "edit");
+
+  const response = await ckEditorApi.getAllConnectedUserIds(ckEditorDocumentId)
+  const connectedUsers = JSON.parse(response);
+  if (connectedUsers.includes(userId)) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+addCronJob({
+  name: 'cleanupStaleCkEditoreUserSessions',
+  interval: 'every 2 hours',
+  async job() {
+    const activeUserSessions = await CkEditorUserSessions.find({endedAt: {$exists: false}}, {sort:{createdAt: -1}}).fetch();
+    const groupedSessions = groupBy(activeUserSessions, session => `${session.userId}-${session.documentId}`);
+
+    const superfluousSessionPromises = [];
+    const latestSessions = []
+
+    for (const sessions of Object.values(groupedSessions)) {
+
+      latestSessions.push(sessions[0])
+      // End all sessions except the most recent one
+      for (let i = 1; i < sessions.length; i++) {
+        superfluousSessionPromises.push(
+          endCkEditorUserSession(sessions[i]._id, "cron")
+        );
+      }
```

Comment:
This should probably be 
```
const superfluousSessions = sessions.slice(1);
superfluousSessionPromises.push(...superfluousSessions.map(session => endCkEditorUserSession(session._id, "cron")));
```

Less chance of off-by-one errors (especially by people reading it).  (But if you don't want to retest it now, can do it later.)